### PR TITLE
Capture user actions times for MV2 benchmark and generate artifacts

### LINKED ISSUE #15282: MV3: capture time for loading account balance in E2E
We need an e2e test that captures how much time it takes from Creating an Account until its Balance is loaded.

### LINKED ISSUE #15290: Capture response time metrics for extension

### LINKED ISSUE #15292: Create build artifacts for response time captured

### PR DESCRIPTION
**Pull request by seaona**
## Explanation
We need a way to measure the delay caused by a service-worker re-start on common user actions along the application.
For that, this PR will take care of measuring common action times, on an MV2 test build, so we have this time reference. Actions covered:

- how much time it takes for loading a newly created account (from Create Account button until balance is loaded)
- how much time it takes from clicking Confirm tx until Tx is confirmed

## More Information
* Closes #15282
* Closes #15290
* Closes #15292

## Manual Testing Steps
Check the build artifacts created on this branch by Metamaskbot, and ensure that there is a file mv3 that captures mv3 e2e actions stats.

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added [n/a]
- [x] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone [n/a]

### + If there are functional changes:

- [ ] Manual testing complete & passed [n/a]
- [ ] "Extension QA Board" label has been applied [n/a]

For testing, check the last MetaMask bot artifacts generated under MV2 E2e Actions Stats.

https://user-images.githubusercontent.com/54408225/183688166-d4c9fd98-83cb-4afb-8381-efaa773c5c85.mp4

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]
LGTM

File: test/e2e/user-actions-benchmark.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+const path = require('path');
+const { promises: fs } = require('fs');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+const { exitWithError } = require('../../development/lib/exit-with-error');
+const {
+  isWritable,
+  getFirstParentDirectoryThatExists,
+} = require('../helpers/file');
+const { convertToHexValue, withFixtures } = require('./helpers');
```
Comment (jpuri) [inline test/e2e/user-actions-benchmark.js:10]: Imports can be organised better.

File: test/e2e/user-actions-benchmark.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      await driver.clickElement({ text: 'Create Account', tag: 'div' });
+      await driver.fill('.new-account-create-form input', '2nd account');
+      await driver.clickElement({ text: 'Create', tag: 'button' });
+      await driver.waitForSelector({
+        css: '.currency-display-component__text',
+        text: '0',
+      });
+      const timestampAfterAction = new Date();
+      loadingTimes = timestampAfterAction - timestampBeforeAction;
+      console.log('end of test');
```
Comment (jpuri) [inline test/e2e/user-actions-benchmark.js:46]: Is above `console.log` intended ?

Comment (seaona) [reply test/e2e/user-actions-benchmark.js:46]: oh nice catch!! removing it right now @jpuri thank you!

File: test/e2e/user-actions-benchmark.js (hunk 1 above)
Comment (seaona) [reply test/e2e/user-actions-benchmark.js]: Ty for the suggestion!! I've sorted them @jpuri

**Review by darkwing — DISMISSED** [review]

File: test/e2e/user-actions-benchmark.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      _yargs.option('out', {
+        description:
+          'Output filename. Output printed to STDOUT of this is omitted.',
+        type: 'string',
+        normalize: true,
+      }),
+  );
+
+  const results = {};
+  results.loadAccount = await loadAccount();
```
Comment (digiwand) [inline test/e2e/user-actions-benchmark.js:119]: would it be helpful if we renamed this prop and the function name to loadNewAccount?

Comment (digiwand) [reply test/e2e/user-actions-benchmark.js:119]: would it be helpful if we renamed this prop and the function name to loadNewAccount or even createAndLoadNewAccount?

**Comment by seaona** [comment]
@digiwand thank you v much for the feedback! I've updated it

**Review by digiwand — DISMISSED** [review]

File: test/e2e/user-actions-benchmark.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      await inputAmount.press(driver.Key.BACK_SPACE);
+      await inputAmount.press(driver.Key.BACK_SPACE);
+      await driver.wait(async () => {
+        const sendDialogMsgs = await driver.findElements(
+          '.send-v2__form div.dialog',
+        );
+        return sendDialogMsgs.length === 1;
+      }, 10000);
+
+      await inputAmount.fill('1');
```
Comment (PeterYinusa) [inline test/e2e/user-actions-benchmark.js:83]: Why do we enter an amount twice? 
It looks like first we enter 1000, clear the field, then enter 1.

Comment (seaona) [reply test/e2e/user-actions-benchmark.js:71]: oh true! I removed the unnecessary steps (for context, this testcase was inherited from the send test file, which does this extra step, but not really needed here).
Thank you @PeterYinusa , fixed!

**Review by digiwand — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**